File: src/main/java/com/bitquabit/kortune/Kortune.java
package com.bitquabit.kortune;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.List;
import java.util.regex.Pattern;

public class Kortune {
    private static final Pattern divider = Pattern.compile("^%$", Pattern.MULTILINE);

    private static String readAll(String resourceName) {
        try {
            BufferedReader reader = new BufferedReader(new InputStreamReader(
                    Kortune.class.getResourceAsStream(resourceName)));
            char[] chars = new char[4096];
            StringBuilder sb = new StringBuilder();
            int read = 0;
            while ((read = reader.read(chars)) > 0) {
                sb.append(chars, 0, read);
            }
            return sb.toString();
        } catch (IOException e) {
            return null;
        }
    }

    private static List<String> loadFortunes(String resourceName) {
        String contents = readAll(resourceName);
        String[] matches = divider.split(contents);
        return Arrays.asList(matches).subList(1, matches.length);
    }

    public static void main(String[] args) {
        List<String> fortunes = loadFortunes("/offensive.fortunes");
        System.out.println(fortunes.get(5));
    }
}

File: src/main/java/com/bitquabit/kortune/Kortune.kt
package com.bitquabit.kortune

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.Random

object Kortune {
    private fun loadFortunes(resourceName: String): List<String> {
        val fortunes = mutableListOf<String>()

        val reader = BufferedReader(InputStreamReader(
                Kortune::class.java.getResourceAsStream(resourceName)))
        val sb = StringBuilder()
        reader.forEachLine { line ->
            if (line == "%") {
                fortunes.add(sb.toString())
                sb.setLength(0)
            } else {
                sb.append(line)
                sb.append("\n")
            }
        }
        fortunes.add(sb.toString())

        return fortunes
    }

    private fun loadFortunes(vararg resourceNames: String): List<String> {
        val fortunes = mutableListOf<String>()
        for (resourceName in resourceNames) {
            fortunes.addAll(loadFortunes(resourceName))
        }
        return fortunes
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val fortunes = loadFortunes("/offensive.fortunes", "/freebsd.fortunes")
        val random = Random()
        println(fortunes[random.nextInt(fortunes.size)])
    }
}
